src/components/modals/ManagePDNTModal/ManagePDNTModal.tsx

```diff
@@ -69,54 +76,76 @@ function ManagePDNTModal() {
       .filter((n) => !!n);
   }
 
-  async function fetchPDNTDetails(txId: ArweaveTransactionID) {
+  async function fetchPDNTDetails(
+    address: ArweaveTransactionID,
+    contractTxId: ArweaveTransactionID,
+  ) {
     try {
       setLoading(true);
-      const names = getAssociatedNames(txId);
-      const [contractState, confirmations] = await Promise.all([
-        arweaveDataProvider.getContractState<PDNTContractJSON>(txId),
-        arweaveDataProvider.getTransactionStatus(txId),
-      ]);
-      const pdnt = new PDNTContract(contractState);
-      setPDNTState(pdnt);
+      const names = getAssociatedNames(contractTxId);
+      const [contractState, confirmations, pendingContractInteractions] =
+        await Promise.all([
+          arweaveDataProvider.getContractState<PDNTContractJSON>(contractTxId),
+          arweaveDataProvider.getTransactionStatus(contractTxId),
+          arweaveDataProvider.getPendingContractInteractions(
+            contractTxId,
+            address.toString(),
+          ),
+        ]);
+      const contract = new PDNTContract(contractState);
       const record = Object.values(pdnsSourceContract.records).find(
-        (r) => r.contractTxId === txId.toString(),
+        (r) => r.contractTxId === contractTxId.toString(),
       );
       const tier = pdnsSourceContract.tiers.history.find(
         (t) => t.id === record?.tier,
       );
       // TODO: add error messages and reload state to row
-      const consolidatedDetails: ManagePDNTRow & any = {
+      const consolidatedDetails: any = {
```

Comment:
why `any`? It would be good to create this `type` or just let TS infer it

Reply:
we should create a type for it, I'll add 👍

Reply:
[c1aa36b](https://github.com/ar-io/pdns-react/pull/214/commits/c1aa36b05878ee9ebbe3c71fe368154b5a1ba4d9)